### PR TITLE
feat(rust): add more tracing data

### PR DESCRIPTION
**Pull request by etorreborre**
This PR adds instruments more code for tracing:

 - The `#[instrument]` attribute has been added to all the service traits at the `CliState` level (note that [we are still missing a few of those interfaces](https://github.com/build-trust/ockam/issues?q=is%3Aissue+is%3Aopen+trait+%22Command+-+refactor%22)).
 
 - More attributes have been added to user event spans: project id/name/route/authority identity etc...
 
 - The root span of the project journey now has a stable span id based on the project id (this avoids some missing span issues when the project trace is updated from several ockam homes).
 
 - Make a stable trace id for the host journey trace (based on the machine MAC + IP address). This way we can still track user events even in case of a full reset or `OCKAM_HOME` change on a given machine.

### CODE REVIEW

File: implementations/rust/ockam/ockam_api/src/authenticator/enrollment_tokens/acceptor.rs [hunk 1]
```diff
@@ -27,6 +27,7 @@ impl EnrollmentTokenAcceptor {
         Self { tokens, members }
     }
 
+    #[instrument(skip_all, fields(otc = otc.to_string(), from = %from))]
```
Comment (davide-baldo) [inline implementations/rust/ockam/ockam_api/src/authenticator/enrollment_tokens/acceptor.rs]: OTC is a sensitive field, if the user is already enrolled, the OTC code won't be consumed.
Maybe we can track the hash of the OTC?

File: implementations/rust/ockam/ockam_api/src/authenticator/enrollment_tokens/acceptor_client.rs [hunk 2]
```diff
@@ -15,6 +15,7 @@ pub trait TokenAcceptor {
 
 #[async_trait]
 impl TokenAcceptor for AuthorityNodeClient {
+    #[instrument(skip_all, fields(token = token.to_string()))]
     async fn present_token(&self, ctx: &Context, token: OneTimeCode) -> miette::Result<()> {
```
Comment (davide-baldo) [inline implementations/rust/ockam/ockam_api/src/authenticator/enrollment_tokens/acceptor_client.rs:19]: Token is a sensitive field, maybe we can track the hash instead?

File: implementations/rust/ockam/ockam_api/src/cloud/enroll.rs [hunk 3]
```diff
@@ -59,6 +60,7 @@ impl Enroll for ControllerClient {
             .into_diagnostic()
     }
 
+    #[instrument(skip_all, fields(enrollment_token = %enrollment_token))]
```
Comment (davide-baldo) [inline implementations/rust/ockam/ockam_api/src/cloud/enroll.rs]: sensitive field, can we use an hash?

File: implementations/rust/ockam/ockam_api/src/enroll/enrollment.rs [hunk 4]
```diff
@@ -71,6 +71,7 @@ impl<T: HasSecureClient + Send + Sync> Enrollment for T {
 // FiXME: this has duplicate with AuthorityNodeClient
 #[async_trait]
 impl Enrollment for SecureClient {
+    #[instrument(skip_all, fields(token = %token))]
```
Comment (davide-baldo) [inline implementations/rust/ockam/ockam_api/src/enroll/enrollment.rs]: sensitive field

File: implementations/rust/ockam/ockam_api/src/enroll/enrollment.rs [hunk 5]
```diff
@@ -93,6 +94,7 @@ impl Enrollment for SecureClient {
         }
     }
 
+    #[instrument(skip_all, fields(token = token.clone().to_string()))]
```
Comment (davide-baldo) [inline implementations/rust/ockam/ockam_api/src/enroll/enrollment.rs]: Sensitive field

File: implementations/rust/ockam/ockam_api/src/enroll/enrollment.rs [hunk 6]
```diff
@@ -107,6 +109,7 @@ impl Enrollment for SecureClient {
             .into_diagnostic()
     }
 
+    #[instrument(skip_all, fields(token = token.clone().to_string()))]
```
Comment (davide-baldo) [inline implementations/rust/ockam/ockam_api/src/enroll/enrollment.rs]: sensitive field

**Review by davide-baldo — DISMISSED** [review]

File: implementations/rust/ockam/ockam_api/src/enroll/enrollment.rs (hunk 6 above)
Comment (etorreborre) [reply implementations/rust/ockam/ockam_api/src/enroll/enrollment.rs]: Done.

File: implementations/rust/ockam/ockam_api/src/enroll/enrollment.rs (hunk 5 above)
Comment (etorreborre) [reply implementations/rust/ockam/ockam_api/src/enroll/enrollment.rs]: Done.

File: implementations/rust/ockam/ockam_api/src/enroll/enrollment.rs (hunk 4 above)
Comment (etorreborre) [reply implementations/rust/ockam/ockam_api/src/enroll/enrollment.rs]: Done.

File: implementations/rust/ockam/ockam_api/src/cloud/enroll.rs (hunk 3 above)
Comment (etorreborre) [reply implementations/rust/ockam/ockam_api/src/cloud/enroll.rs]: Done.

File: implementations/rust/ockam/ockam_api/src/authenticator/enrollment_tokens/acceptor_client.rs (hunk 2 above)
Comment (etorreborre) [reply implementations/rust/ockam/ockam_api/src/authenticator/enrollment_tokens/acceptor_client.rs:19]: Done.

File: implementations/rust/ockam/ockam_api/src/authenticator/enrollment_tokens/acceptor.rs (hunk 1 above)
Comment (etorreborre) [reply implementations/rust/ockam/ockam_api/src/authenticator/enrollment_tokens/acceptor.rs]: Done.

**Comment by etorreborre** [comment]
Thanks @davide-baldo, I just removed those fields for now.

**Review by davide-baldo — APPROVED** [review]

**Pull request merged**